sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/JsonDocumentShredder.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.opentelemetry.extension.annotations.WithSpan;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCode;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCodeRuntimeException;
+import io.stargate.sgv2.docsapi.config.constants.Constants;
+import io.stargate.sgv2.docsapi.service.util.DocsApiUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class JsonDocumentShredder {
+
+  private final DocumentProperties properties;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public JsonDocumentShredder(DocumentProperties properties, ObjectMapper objectMapper) {
+    this.properties = properties;
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * Shreds the JSON payload and returns the list of {@link JsonShreddedRow} for each value that
+   * should be stored in the data store.
+   *
+   * @param payload JSON payload as string
+   * @param subDocumentPath Prefix path to use. Note that paths are added to each row path as they
+   *     are given, without any modifications.
+   * @return List of shredded rows
+   */
+  public List<JsonShreddedRow> shred(String payload, List<String> subDocumentPath) {
```

Comment:
In v1, `DocumentWriteService` contains this comment:
```java
    // TODO this smells like a bad design
    //  for v2 we need to not include sub-path to shredding, and cover it here as a wrapper
    checkPathMatchesRows(subDocumentPath, rows);
```
Does that mean we should remove `subDocumentPath` here?

Reply:
I guess that was my idea, but this can be done later on.. It's a small optimization..

@EricBorczuk can you create a ticket please?

Reply:
Sounds good, then I'll preserve the TODO in #1833.